Source/RK3/EddyViscosity.cpp

```diff
@@ -0,0 +1,115 @@
+// Generated by Pankaj Jha on 05/31/2021
+#include <RK3.H>
+
+using namespace amrex;
+
+// Compute Eddy Viscosity
+Real
+ComputeTurbulentViscosity(
+  const int& i,
+  const int& j,
+  const int& k,
+  const Array4<Real>& u,
+  const Array4<Real>& v,
+  const Array4<Real>& w,
+  const NextOrPrev& nextOrPrev,
+  const MomentumEqn& momentumEqn,
+  const DiffusionDir& diffDir,
+  const Geometry& geom,
+  Array4<Real>& nut)
+{
+  return 0;
+}
+
+/// Compute K (i-1/2, j+1/2, k) etc given K(i, j, k) or nut (i, j, k) is known
+Real
+InterpolateTurbulentViscosity(const int &i, const int &j, const int &k,
+                              const Array4<Real>& u, const Array4<Real>& v, const Array4<Real>& w,
+                              const enum NextOrPrev &nextOrPrev,
+                              const enum MomentumEqn &momentumEqn,
+                              const enum DiffusionDir &diffDir,
+                              const Geometry &geom,
+                              const Array4<Real>& nut) {
+  // Assuming we already have 'nut' computed for all (i, j, k)
+// TODO: Check exactly where we should compute 'nut'
+  Real turbViscInterpolated = 1.0;
+
+  switch (momentumEqn) {
+  case MomentumEqn::x:
+    switch (diffDir) {
+    case DiffusionDir::x:
+      if (nextOrPrev == NextOrPrev::next)
+        turbViscInterpolated = nut(i, j, k); // K (i, j, k) needed to obtain tau11 (i+1/2)
```

Comment:
@pkjha-aero Is it possible to refactor this code and make a subroutine to precalculate the indexes before calling the nut routine at the end? This would allow reuse of the switch case logic at multiple places in the code.

Reply:
@rfhaque I'll take care of this in future PR.